File: clients/algoliasearch-client-ruby/lib/algolia/chunked_helper_options.rb
module Algolia
  # Optional configuration for chunked helpers that batch records and poll for task completion.
  class ChunkedHelperOptions
    DEFAULT_MAX_RETRIES = 100
    DEFAULT_REPLACE_ALL_OBJECTS_MAX_RETRIES = 800
    attr_reader :max_retries

    def initialize(max_retries: DEFAULT_MAX_RETRIES)
      @max_retries = max_retries
    end

    def self.resolve(options)
      options || new
    end
  end
end

File: clients/algoliasearch-client-kotlin/client/src/commonMain/kotlin/com/algolia/client/extensions/ChunkedHelperOptions.kt
package com.algolia.client.extensions

/** The default maximum number of retries for chunked helpers polling for task completion. */
public const val DEFAULT_MAX_RETRIES: Int = 100

/**
 * Optional configuration shared across chunked helpers that batch records and poll for task
 * completion.
 *
 * @param maxRetries The maximum number of retries when polling for task completion. Defaults to
 *   [DEFAULT_MAX_RETRIES].
 */
public data class ChunkedHelperOptions(
  public val maxRetries: Int = DEFAULT_MAX_RETRIES,
)
